packages/ui/src/components/SelectInputV2/Dropdown.tsx

```diff
@@ -738,16 +748,33 @@ export const Dropdown = ({
       setSearch('')
     }
 
-    document.addEventListener('mousedown', event =>
-      handleClickOutside(
-        event,
-        ref,
-        setIsDropdownVisible,
-        refSelect,
-        onSearch,
-        options,
-      ),
-    )
+    const modalElement = document.getElementById('backdrop-modal')
+
+    if (modalElement) {
+      modalElement.addEventListener('mousedown', event =>
+        handleClickOutside(
+          event,
+          ref,
+          setIsDropdownVisible,
+          refSelect,
+          onSearch,
+          options,
+          true,
+        ),
+      )
+    } else {
+      document.addEventListener('mousedown', event =>
+        handleClickOutside(
+          event,
+          ref,
+          setIsDropdownVisible,
+          refSelect,
+          onSearch,
+          options,
+          false,
+        ),
+      )
+    }
```

Comment:
Same here the select input should have no knowledge about the modal. This is an issue that the modal has more than select input